### PR TITLE
Add missing links in platform support docs

### PR DESCRIPTION
**Pull request by Michcioperz**
I was looking at m68k support and saw that https://doc.rust-lang.org/rustc/platform-support.html and the sidebar there were missing some links to target documentation

### CODE REVIEW

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Looks good to me, thanks! I'll let someone from the compiler team make the final approval.

r? @pnkfelix

File: src/doc/rustc/src/SUMMARY.md [hunk 1]
```diff
@@ -15,13 +15,15 @@
 - [Platform Support](platform-support.md)
     - [Template for target-specific documentation](platform-support/TEMPLATE.md)
     - [aarch64-apple-ios-sim](platform-support/aarch64-apple-ios-sim.md)
+    - [aarch64-unknown-none-hermitkernel](platform-support/aarch64-unknown-none-hermitkernel.md)
     - [armv7-unknown-linux-uclibceabi](platform-support/armv7-unknown-linux-uclibceabi.md)
     - [armv7-unknown-linux-uclibceabihf](platform-support/armv7-unknown-linux-uclibceabihf.md)
-    - [aarch64-unknown-none-hermitkernel](platform-support/aarch64-unknown-none-hermitkernel.md)
     - [\*-kmc-solid_\*](platform-support/kmc-solid.md)
```
Comment (Urgau) [inline src/doc/rustc/src/SUMMARY.md:21]:
```suggestion
    - [*-kmc-solid_*](platform-support/kmc-solid.md)
```
It's not rendering them correctly with `\` before the star.

Comment (Michcioperz) [reply src/doc/rustc/src/SUMMARY.md:21]: Tested it locally on mdBook 0.4.15 it doesn't seem to solve it. I tried a few more combinations and it seems that this works:
```suggestion
    - [\*-kmc-solid_\\*](platform-support/kmc-solid.md)
```
as well as
```suggestion
    - [\\*-kmc-solid_\*](platform-support/kmc-solid.md)
```
but I'm not seeing a working option with equal number of `\` or no `\` at all

Comment (pnkfelix) [reply src/doc/rustc/src/SUMMARY.md:21]: @Michcioperz can you clarify whether one of the suggestions you have written here should be adopted?

**Comment by pnkfelix** [comment]
r=me once the author has resolved the question about the `[\*-kmc-solid_\*]` thing

File: src/doc/rustc/src/SUMMARY.md (hunk 1 above)
Comment (ehuss) [reply src/doc/rustc/src/SUMMARY.md:21]: This is a bug with mdbook.  I posted a fix at https://github.com/rust-lang/mdBook/pull/1785.  I recommend leaving it as the original `[\*-kmc-solid_\*]`.  I'll publish a new version of mdbook soon and update it here and that should fix the rendering.

**Comment by ehuss** [comment]
Thanks! I'm going to go ahead and approve. The rendering issue should be resolved by #96088

@bors r=pnkfelix rollup

**Pull request merged**